app/tasks.py

```diff
@@ -437,3 +438,19 @@ def _should_notify(self, count_of_unreviewed_notes):
         return  count_of_unreviewed_notes > self.config.get(
                 'unreviewed_notes_threshold')
 
+
+class ThumbnailPreparer(utils.BaseHandler):
+
+    repo_required = False
+    ACTION = 'tasks/thumbnail_preparer'
+
+    def get(self):
+        if self.repo:
+            for p in (model.Photo.all()
```

Comment:
It may be rare, but this may not finish within task queue timeout (10 min?). Can you handle timeout and spawn a new task to continue the task? See https://github.com/google/personfinder/blob/23d7030eec886204c06598e5403700aa11dad16d/app/tasks.py#L86 for reference. It may be nice to define a common subclass to deal with the retry-on-timeout logic.

Reply:
I think it's actually ok if it doesn't finish, since the cron job's next run will start soon anyway. And automatic restarts would risk having two tasks running at once, which is kinda messy since they'd both be trying to do the same thing.

Reply:
Oh I see, so the key here is that you set `filter('thumbnail_data =', None)` so records which it has generated a thumbnail for will be skipped in the next task.

Can you add a comment explaining why it has no (explicit) retry logic?